### ISSUE TITLE
Parentheses should hug when destructuring array params

### ISSUE THREAD

**Issue by jwbay** [prettier/prettier#3046]
**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#%7B%22content%22%3A%22promise.then((%5Cn%20%20%5Baaaaaa%2C%20bbbbbb%2C%20ccccccc%2C%20dddddddd%2C%20eeeeeee%2C%20ffffffff%2C%20gggggggg%2C%20hhhhhhh%5D%5Cn)%20%3D%3E%20%7B%5Cn%20%20console.log(eeeee)%3B%5Cn%7D)%3B%5Cn%22%2C%22options%22%3A%7B%22ast%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%2C%22jsxBracketSameLine%22%3Afalse%2C%22output2%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22printWidth%22%3A80%2C%22semi%22%3Atrue%2C%22singleQuote%22%3Afalse%2C%22tabWidth%22%3A2%2C%22trailingComma%22%3A%22none%22%2C%22useTabs%22%3Afalse%7D%7D)

**Input:**
```jsx
promise.then((
  [aaaaaa, bbbbbb, ccccccc, dddddddd, eeeeeee, ffffffff, gggggggg, hhhhhhh]
) => {
  console.log(eeeee);
});

```

**Output:**
```jsx
promise.then(
  (
    [aaaaaa, bbbbbb, ccccccc, dddddddd, eeeeeee, ffffffff, gggggggg, hhhhhhh]
  ) => {
    console.log(eeeee);
  }
);

```

**Expected behavior:**
Input preserved, probably as long as it's the only parameter

**Comment by duailibe** [comment]
If we did enable hug, the output would actually be:

```js
promise.then(
  ([
    aaaaaa,
    bbbbbb,
    ccccccc,
    dddddddd,
    eeeeeee,
    ffffffff,
    gggggggg,
    hhhhhhh
  ]) => {
    console.log(eeeee);
  }
);
```
to prevent exceeding printWidth so I guess the fix is actually printing the above instead of:

```
promise.then(
  (
    [
      aaaaaa,
      bbbbbb,
      ccccccc,
      dddddddd,
      eeeeeee,
      ffffffff,
      gggggggg,
      hhhhhhh
    ]
  ) => {
    console.log(eeeee);
  }
);
```

**Comment by duailibe** [comment]
That wouldn't quite match how we break when exceeding print width with normal arguments:

```js
promise.then(
  (
    aaaaaa,
    bbbbbb,
    ccccccc,
    dddddddd,
    eeeeeee,
    ffffffff,
    gggggggg,
    hhhhhhh
  ) => {
    console.log(eeeee);
  }
);
```

So I think we could hug there, only the output would be:
```js
promise.then(
  ([
    aaaaaa,
    bbbbbb,
    ccccccc,
    dddddddd,
    eeeeeee,
    ffffffff,
    gggggggg,
    hhhhhhh
  ]) => {
    console.log(eeeee);
  }
);
```
instead of:
```js
promise.then(
  (
    [
      aaaaaa,
      bbbbbb,
      ccccccc,
      dddddddd,
      eeeeeee,
      ffffffff,
      gggggggg,
      hhhhhhh
    ]
  ) => {
    console.log(eeeee);
  }
);
```

**Comment by duailibe** [comment]
Fixed in 1.9.2!
